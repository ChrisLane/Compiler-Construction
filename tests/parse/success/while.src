// This is to test the while loop expression. This should parse.
main() {
    while (a == 2) {
        a = a + 1;
    };
}
