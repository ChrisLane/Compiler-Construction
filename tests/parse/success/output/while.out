
main (  ) { 
    While ( Operator ( Equal, Identifier "a" , Const 2 ) ) { 
        Asg ( Identifier "a" := Operator ( Plus, Identifier "a" , Const 1 ) ) 
    } 
}

